streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -250,7 +252,7 @@ private void emitNonJoinedOuterRecords(
                     }
 
                     final VOut nullJoinedValue;
-                    if (isLeftSide) {
+                    if (joinSide.isLeftSide()) {
```

Comment:
This is another unsafe place that should be part of the enum.
Try and eliminate all of the `unchecked` warnings in this class.